### PR TITLE
New package: CalculiX-2.20

### PR DESCRIPTION
**Pull request by Phicem**
### NEW PACKAGE - CalculiX

This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

CalculiX is a 3D finite element program. It is an optional but important dependency of FreeCAD (for mechanical and thermal calculations). This PR is related to issue https://github.com/void-linux/void-packages/issues/5603.

CalculiX is split in two parts: the solver (CalculiX-ccx) and the graphical interface and pre/post-processing (CalculiX-cgx).

This PR includes 4 packages, each in a separate commit:
- spooles (build dependency for CalculiX-ccx)
- CalculiX-ccx
- CalculiX-cgx (including tetgen binary)
- CalculiX meta-package (which brings CalculiX-ccx and CalculiX-cgx)

#### Local build testing
- Native x86_64 (glibc and musl): **OK**
- Native i686 (glibc) (from an x86_64 machine) : **OK**
- Native armv7l (glibc): **OK**

#### Cross build testing (from x86_64 machine)
- Cross-compiling to armv7l: **OK** (requires using `./xbps-src binary-boostrap i686` on an x86_64 system, otherwise it fails)
- Cross-compiling to aarch64: **OK**

#### Runtime testing
- On x86_64 (glibc): **OK** (for CalculiX-ccx: some calculation tests differ from the official reference results files)
- On armv7l: **OK** (for CalculiX-ccx: some calculation differences too)
- On i686: **NOT TESTED**

### CODE REVIEW

File: srcpkgs/CalculiX-ccx/template [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# Template file for 'CalculiX-ccx'
+pkgname=CalculiX-ccx
+version=2.20
+revision=1
+archs=~"*-musl"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:5]: why no musl? (here and cgx)

File: srcpkgs/tetgen/template [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+version=1.5.1
+revision=1
+build_style="gnu-makefile"
+make_cmd="make -C ${pkgname}${version} -f makefile"
+hostmakedepends="tar"
+short_desc="Quality Tetrahedral Mesh Generator and 3D Delaunay Triangulator"
+maintainer="Phicem <phicem@gmx.com>"
+license="AGPL-3.0-or-later"
+homepage="https://www.wias-berlin.de/software/index.jsp?id=TetGen"
+distfiles="http://www.dhondt.de/cgx_2.20.all.tar.bz2"
```
Comment (classabbyamp) [inline srcpkgs/tetgen/template:12]: this should probably be a subpackage of cgx

**Review by classabbyamp — COMMENTED** [review]
just a couple of things i noticed. there's a lot of work to be done here, I think, but I need to look into this more

File: srcpkgs/CalculiX-ccx/template (hunk 1 above)
Comment (Phicem) [reply srcpkgs/CalculiX-ccx/template:5]: Cross-compiling from x86_64 to x86_64-musl failed. Since I could not test this on native x86_64-musl, I don't know if this package works on musl systems.

Comment (paper42) [reply srcpkgs/CalculiX-ccx/template:5]: x86_64-musl is not cross compiled, create a musl masterdir instead

Comment (Phicem) [reply srcpkgs/CalculiX-ccx/template:5]: Oops I missed that part of the manual, thank you very much. It seems to work! I will update the package.

File: srcpkgs/tetgen/template (hunk 2 above)
Comment (Phicem) [reply srcpkgs/tetgen/template:12]: Tetgen is actually a separate program, with different authors. It is distributed by tetgen website, but it is necessary to provide an email address before getting the download link. Tetgen is also distributed by CalculiX authors, this time with no email constraint. This is why I picked CalculiX website as the tetgen source.
What do you mean by "subpackage"? Should I call the tetgen package 'CalculiX-tetgen' ?

File: srcpkgs/CalculiX-ccx/template (hunk 1 above)
Comment (Phicem) [reply srcpkgs/CalculiX-ccx/template]: PR updated, I removed the "archs" line.

**Comment by Phicem** [comment]
Changes:
- Removed the musl restriction
- Fixed xlint warnings
- Disabled tests for tetgen (because no tests are provided)

**Comment by Phicem** [comment]
Changes:
- Fixed residual xlint warnings on tetgen

File: srcpkgs/CalculiX-ccx/template [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+_LIBS=" ${XBPS_CROSS_BASE}/usr/lib/spooles.a ${XBPS_CROSS_BASE}/usr/lib/libarpack.so ${XBPS_CROSS_BASE}/lib/libopenblas.so  -lpthread -lm -lc"
+hostmakedepends="perl gcc-fortran"
+makedepends="SPOOLES arpack-ng-devel libgomp-devel openblas-devel"
+_FFLAGS=" -Wall -O2 -fallow-argument-mismatch"
+
+if [ "$XBPS_MACHINE" == "x86_64" ]  && [ "$XBPS_TARGET_MACHINE" == "i686" ]; then
+	makedepends+=" libquadmath"
+	hostmakedepends+=" libgomp-devel-32bit"
+fi
+short_desc="3D Structural Finite Element Program - Solver part"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:16]:
```suggestion
short_desc="3D Structural Finite Element Program - Solver"
```

File: srcpkgs/CalculiX-ccx/template [hunk 4]
```diff
@@ -0,0 +1,35 @@
+# Template file for 'CalculiX-ccx'
+pkgname=CalculiX-ccx
+version=2.20
+revision=1
+wrksrc=CalculiX
+_ccx_dir=ccx_${version}
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:6]:
```suggestion
build_wrksrc="ccx_${version}/src"
```

File: srcpkgs/CalculiX-ccx/template [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+_LIBS=" ${XBPS_CROSS_BASE}/usr/lib/spooles.a ${XBPS_CROSS_BASE}/usr/lib/libarpack.so ${XBPS_CROSS_BASE}/lib/libopenblas.so  -lpthread -lm -lc"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:7]:
```suggestion
```

out of order and unnecessary

File: srcpkgs/CalculiX-ccx/template [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+hostmakedepends="perl gcc-fortran"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:8]:
```suggestion
hostmakedepends="gcc-fortran"
```
perl script is unnecessary, we'll do it manually in a better way

File: srcpkgs/CalculiX-ccx/template [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+makedepends="SPOOLES arpack-ng-devel libgomp-devel openblas-devel"
+_FFLAGS=" -Wall -O2 -fallow-argument-mismatch"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:10]:
```suggestion
```
partially unnecessary and it should be put later

File: srcpkgs/CalculiX-ccx/template [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+if [ "$XBPS_MACHINE" == "x86_64" ]  && [ "$XBPS_TARGET_MACHINE" == "i686" ]; then
+	makedepends+=" libquadmath"
+	hostmakedepends+=" libgomp-devel-32bit"
+fi
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:15]:
```suggestion
```
this should not need to be specified, [i686 is built natively](https://github.com/void-linux/void-packages#building-32bit), not on cross. it's also out of order

File: srcpkgs/CalculiX-ccx/template [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+short_desc="3D Structural Finite Element Program - Solver part"
+maintainer="Phicem <phicem@gmx.com>"
+license="GPL-2.0-or-later"
+homepage="http://www.dhondt.de/"
+distfiles="http://www.dhondt.de/ccx_${version}.src.tar.bz2"
+checksum=63bf6ea09e7edcae93e0145b1bb0579ea7ae82e046f6075a27c8145b72761bcf
+CFLAGS=" -Wall -O2  -I ${XBPS_CROSS_BASE}/usr/include/SPOOLES-2.2 -DARCH='Linux' -DSPOOLES -DARPACK -DMATRIXSTORAGE -DNETWORKOUT"
+CFLAGS+=" ${XBPS_CROSS_CFLAGS}"
+_FFLAGS+=" ${XBPS_CROSS_CFLAGS}"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:24]:
```suggestion

FFLAGS="-fallow-argument-mismatch"
CFLAGS="-I${XBPS_CROSS_BASE}/usr/include/SPOOLES -DARCH=Linux -DSPOOLES -DARPACK -DMATRIXSTORAGE -DNETWORKOUT"
```
`xbps-src` integrates the CFLAGS/FFLAGS specified at the top level of the template into its own, and then exposes that to the packaging phases, so we can trim this down

File: srcpkgs/CalculiX-ccx/template [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:25]:
```suggestion

post_patch() {
	vsed -e 's,./date.pl; ,,' -i Makefile
	# taken from date.pl, changed for reproducibility
	vsed -e 's/You are using an executable made on.*/You are using an executable made on '"$(date --utc --date @$SOURCE_DATE_EPOCH +%Y-%m-%d)"'\\n");/g' \
		-i ccx_$version.c -i CalculiXstep.c
	vsed -e 's/COMPILETIME.*/COMPILETIME       '"$(date --utc --date @$SOURCE_DATE_EPOCH +%Y-%m-%d)"'\\n",p1);/g' \
		-i frd.c
}

```
instead of `date.pl` (which is broken, there's an incorrect filename), we can do the same thing manually, and at the time use a more reproducible method of getting the date.

File: srcpkgs/CalculiX-ccx/template [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+do_build() {
+	if [ "$CROSS_BUILD" ]; then
+		_xbps_cross_prefix="${XBPS_CROSS_TRIPLET}-"
+	fi
+	make -C ${_ccx_dir}/src CC="${_xbps_cross_prefix}gcc" FC="${_xbps_cross_prefix}gfortran" LIBS="$_LIBS" FFLAGS="$_FFLAGS" CFLAGS="$CFLAGS"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:30]:
```suggestion
	make CC="$CC" FC="$FC" CFLAGS="$CFLAGS" FFLAGS="$FFLAGS" \
		LIBS="${XBPS_CROSS_BASE}/usr/lib/spooles.a ${XBPS_CROSS_BASE}/usr/lib/libarpack.so ${XBPS_CROSS_BASE}/usr/lib/libopenblas.so -lpthread -lm -lc"
```
no need for all the special handling you were doing, `xbps-src` provides those proper values for you

File: srcpkgs/CalculiX-ccx/template [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+do_install() {
+	vbin ${_ccx_dir}/src/ccx_${version} ccx
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-ccx/template:34]:
```suggestion
	vbin ccx_${version} ccx
```

File: srcpkgs/CalculiX-cgx/files/README_LICENSES [hunk 13]
```diff
@@ -0,0 +1,5 @@
+cgx program is released under GNU GPL v2 or later (except for a few functions, see CGX_LICENSE)
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/files/README_LICENSES:1]: I don't think this file is necessary. having the licences installed should be enough

File: srcpkgs/CalculiX-cgx/template [hunk 14]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'CalculiX-cgx'
+pkgname=CalculiX-cgx
+version=2.20
+revision=1
+wrksrc=CalculiX
+_cgx_dir=cgx_${version}
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/template:6]:
```suggestion
create_wrksrc=yes
build_wrksrc="CalculiX/cgx_${version}/src"
```
this ensures there's nothing left in the builddir on clean

File: srcpkgs/CalculiX-cgx/patches/Makefile.patch [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+   -I./ \
+@@ -67,7 +68,7 @@
+ OUTIL = $(SUTIL:.c=.o)
+ 
+ cgx: $(OLIB) $(OUTIL) $(OULIB)
+-	g++  $(OULIB)  $(OLIB) $(OUTIL)  $(CFLAGS) $(LFLAGS) -o  $@
++	${CXX}  $(OULIB)  $(OLIB) $(OUTIL)  $(CFLAGS) $(LFLAGS) -o  $@
+ 
+ 
+
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/patches/Makefile.patch:16]:
```suggestion
--- a/CalculiX/cgx_2.20/src/Makefile
+++ b/CalculiX/cgx_2.20/src/Makefile
@@ -67,7 +67,7 @@
 OUTIL = $(SUTIL:.c=.o)
 
 cgx: $(OLIB) $(OUTIL) $(OULIB)
-	g++  $(OULIB)  $(OLIB) $(OUTIL)  $(CFLAGS) $(LFLAGS) -o  $@
+	$(CXX)  $(OULIB)  $(OLIB) $(OUTIL)  $(CFLAGS) $(LFLAGS) -o  $@
 
 
 

```
only this much of the patch should be needed

File: srcpkgs/CalculiX-cgx/template [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+build_style="gnu-makefile"
+make_build_args="-C ${_cgx_dir}/src"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/template:8]:
```suggestion
```
I ended up overriding all the useful parts of the build style so I don't think it's needed

File: srcpkgs/CalculiX-cgx/template [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+depends="tetgen"
+short_desc="3D Structural Finite Element Program - Graphical interface"
+maintainer="Phicem <phicem@gmx.com>"
+license="GPL-2.0-or-later, custom:GLUT"
+homepage="http://www.dhondt.de/"
+distfiles="http://www.dhondt.de/cgx_${version}.all.tar.bz2"
+checksum=f4c840c3633718547e33f86b2d7bc6b35608dd8d6cbf9fa192cecef34f2dc908
+if [ "$XBPS_MACHINE" == "x86_64" ] ; then
+	hostmakedepends+="glibc-devel-32bit"
+fi
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/template:19]:
```suggestion
```
not needed

File: srcpkgs/CalculiX-cgx/files/README_LICENSES [hunk 18]
```diff
@@ -0,0 +1,5 @@
...
+
+To be compiled, this program makes use of other programs released under different licenses:
+* glut is released under a custom license (see GLUT_LICENSE)
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/files/README_LICENSES:4]: the glut licence's status as free software is dubious. maybe this should be `repository=nonfree`, not sure

File: srcpkgs/CalculiX-cgx/template [hunk 19]
```diff
@@ -0,0 +1,27 @@
...
+CFLAGS=" -O2 -Wall -Wno-narrowing  -I./  -I/usr/include/  -I/usr/include/GL  -I../../libSNL/src  -I../../glut-3.5/src  -I/usr/X11/include"
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/template:20]:
```suggestion
CFLAGS="-DSEMINIT -I. -I${XBPS_CROSS_BASE}/usr/include/GL -I${XBPS_CROSS_BASE}/usr/X11/include -I../../libSNL/src -I../../glut-3.5/src"
```
only need to include the special cflags

File: srcpkgs/CalculiX-cgx/template [hunk 20]
```diff
@@ -0,0 +1,27 @@
...
+
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/template:21]:
```suggestion

do_build() {
	make CC=$CC CXX=$CXX CFLAGS="$CFLAGS"
}

```

File: srcpkgs/CalculiX-cgx/template [hunk 21]
```diff
@@ -0,0 +1,27 @@
...
+do_install() {
+	vbin ${_cgx_dir}/src/cgx
+	vlicense ${_cgx_dir}/README CGX_LICENSE
+	vlicense glut-3.5/NOTICE GLUT_LICENSE
+	vlicense ${FILESDIR}/README_LICENSES
```
Comment (classabbyamp) [inline srcpkgs/CalculiX-cgx/template:26]:
```suggestion
	vbin cgx
	vlicense ../README CGX_LICENSE
	vlicense ../../glut-3.5/NOTICE GLUT_LICENSE
```

**Review by classabbyamp — COMMENTED** [review]
ok I think that's everything that I can see

File: srcpkgs/SPOOLES/template [hunk 22]
```diff
@@ -0,0 +1,25 @@
+# Template file for 'SPOOLES'
+pkgname=SPOOLES
```
Comment (ahesford) [inline srcpkgs/SPOOLES/template:2]: This package should be called `spooles`.

File: srcpkgs/tetgen/template (hunk 2 above)
Comment (ahesford) [reply srcpkgs/tetgen/template:12]: Yeah, we don't play this game with Void packages.

You get the tarball from the proper upstream source, unless it is explicitly vendored into the rest of CalculiX, in which case it wouldn't be a separate package at all.

If this is difficult or impossible, it militates strongly against packaging, not grabbing upstream distfiles from some other distributor.

File: srcpkgs/CalculiX-cgx/files/README_LICENSES (hunk 18 above)
Comment (0x5c) [reply srcpkgs/CalculiX-cgx/files/README_LICENSES:4]: Reading the GLUT licence, it seems to me that it's essentially a MIT-like permissive licence

Comment (classabbyamp) [reply srcpkgs/CalculiX-cgx/files/README_LICENSES:4]: ok that seems right to me

**Comment by Phicem** [comment]
Thank you for all your comments! I marked the PR as draft until I test all these modifications.

File: srcpkgs/tetgen/template (hunk 2 above)
Comment (Phicem) [reply srcpkgs/tetgen/template:12]: Tetgen is indeed officially vendored along with CalculiX (more precisely with the cgx part). It is explicitely written in the documentation.

Initially I wanted to put tetgen a separate package in case someone wants to use tetgen without installing CalculiX-cgx. But it is probably a rare use-case, and Calculix-cgx package is not that huge (15 MB).

If that sounds good to you, I will include tetgen binary directly into CalculiX-cgx package.

File: srcpkgs/CalculiX-cgx/template [hunk 23]
```diff
@@ -0,0 +1,27 @@
+# Template file for 'CalculiX-cgx'
```
[the remark this answers is not in the document]
Comment (Phicem) [reply srcpkgs/CalculiX-cgx/template:1]: No, cgx can be used without ccx (and ccx can be used without cgx as well).

File: srcpkgs/tetgen/template (hunk 2 above)
Comment (Phicem) [reply srcpkgs/tetgen/template:12]: However if I do that (including tetgen binary directly into CalculiX-cgx package), I'm afraid I will have to define a custom do_build() function, since there are two independant makefile/cmake files. I could patch cgx makefile so that it calls tetgen makefile, but I don't find this very elegant.

An alternative is to keep tetgen in a separate sub-package called CalculiX-tetgen, using the tetgen version vendored by CalculiX.

In short, the solutions I see are:
1. single package with a custom do_build() function
2. single package with a patch on the makefile
3. keeping a separate subpackage called Calculix-tetgen

I don't know which one is the cleanest option. What do you think?

File: srcpkgs/SPOOLES/template (hunk 22 above)
Comment (Phicem) [reply srcpkgs/SPOOLES/template:2]: I too find the uppercase name ugly, but the vendor uses uppercase letters when they speak about their software. The folder name after tgz extraction is also uppercase. So I followed the xbps-src manual recommandation ("As a rule of thumb, if the tar.gz of a package contains uppercase letter, then the package name should contain them too").

I think the reason why the name is uppercase is that the library name is an acronym (for "SParse Object Oriented Linear Equations Solver").

So in my opinion the package name should be kept uppercase. Do you agree?

File: srcpkgs/tetgen/template (hunk 2 above)
Comment (0x5c) [reply srcpkgs/tetgen/template:12]: If you need to give your email to access the download, it sounds likely to me that the licence might also be a problem in itself

Comment (Phicem) [reply srcpkgs/tetgen/template:12]: The authors say that tetgen is distributed "as Free Software under the [GNU Affero Public License v. 3.0](http://www.gnu.org/licenses/agpl-3.0.html)". But they also sell a commercial license for those who need it.

The GNU Affero Public License v. 3.0 says that "You are not required to accept this License in order to receive or run a copy of the Program", which is contradictory with the fact that they ask the user to give an email address and accept the license before getting the download link. So my understanding is that they don't really respect the license themselves (but I guess this is not a problem since they are the authors).

Comment (0x5c) [reply srcpkgs/tetgen/template:12]: Yeah it's not a problem for them if they are the copyright holders of 100% of the code (and I assume they are, in order to dual-licence).

Anyway it's some form of GPL so my potential concerns about the licence are addressed

File: srcpkgs/CalculiX-cgx/template (hunk 17 above)
Comment (Phicem) [reply srcpkgs/CalculiX-cgx/template:19]: I added these lines during my attempts to cross-compile to armv7l (from an x86_64 machine). Without them, I get a compiler error about a missing header:
`fatal error: gnu/stubs-32.h: No such file or directory`

Adding glibc-devel-32bit fixes this issue, but is not enough to make compilation work, because then there is another error:
`/usr/include/math.h:462:21: error: '_Float128' is not supported on this target`

Since cross-compilation does not work, I guess you're right and these lines are not needed.

File: srcpkgs/tetgen/template [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+checksum=f4c840c3633718547e33f86b2d7bc6b35608dd8d6cbf9fa192cecef34f2dc908
+make_check="no" # No tests provided
+
+do_extract() {
+	local srcdir="$XBPS_SRCDISTDIR/$pkgname-$version"
+	tar -xvjf $srcdir/"$(basename ${distfiles})"
+	pwd
+	tar -xvf ${pkgname}${version}.tar
```
[the remark this answers is not in the document]
Comment (Phicem) [reply srcpkgs/tetgen/template:20]: Yes it's much better.

File: srcpkgs/tetgen/template (hunk 2 above)
Comment (ahesford) [reply srcpkgs/tetgen/template:11]: "Vendored" in this context means that the second project is integrated into the source tree and built alongside it. It seems what CalculiX is doing is just co-distributing tetgen, not vendoring it---the tarball is included but the CalculiX build process has no knowledge of this.

If you can resolve the licensing issue and get the original tarball from the original tetgen source, a separate package is OK. Otherwise, build it into the CalculiX package. You can override `do_build` or define `pre_build` or `post_build` as appropriate.

I don't see much point in making tetgen a subpackage, and that doesn't resolve the build override problem anyway---subpackage installation steps are not meant to add additional build steps, just to split installed contents.

File: srcpkgs/SPOOLES/template (hunk 22 above)
Comment (Phicem) [reply srcpkgs/SPOOLES/template]: Subtle. Let's go with `spooles` then.

File: srcpkgs/SPOOLES/template [hunk 25]
```diff
@@ -0,0 +1,25 @@
...
+
+do_install() {
+	vinstall spooles.a 755 usr/lib
+	HEADERS_DIR=usr/include/SPOOLES-2.2
+	for file in $(find -name '*.h') ; do
+		dir=$(dirname ${file})
+		vmkdir ${HEADERS_DIR}/${dir}
+		vinstall ${file} 644 ${HEADERS_DIR}/${dir}
+	done
+
```
[the remark this answers is not in the document]
Comment (Phicem) [reply srcpkgs/SPOOLES/template]: Yes your version is better. Commited.

File: srcpkgs/CalculiX-cgx/files/README_LICENSES (hunk 13 above)
Comment (Phicem) [reply srcpkgs/CalculiX-cgx/files/README_LICENSES:1]: I admit this file has limited interest. I wanted to make explicit the fact that libSN is released under GNU GPL v2 (or later), which I did not find that obvious given the multiple licenses of the package. But since the package license is GPL v2 + custom  and there is no license file for libSN, I guess it implicitely means that libSN is released under GPL v2.

Thus I will remove this file as you suggest.

File: srcpkgs/CalculiX-cgx/template (hunk 19 above)
Comment (Phicem) [reply srcpkgs/CalculiX-cgx/template:17]: Hmm, I think I understand most of the suggested change here, but not all of it.

I understand that the `-02` flag is useless because it is already defined in `$XBPS_CFLAGS` and thus implicitely used.

I guess removing `-Wall` only suppresses warnings (which are not very useful in an automated build process), and can be safely removed, so I think I undestand this one as well. (I am correct?)

However I don't understand the removal of the `-Wno-narrowing` flag. I'm not very familiar with this flag but I thought it could remove errors (and not only warnings), and thus could change the build result. Why did you suggest removing this flag? Is it because it is considered that software should not be built when there are narrowing errors? Or is there another reason?

Comment (classabbyamp) [reply srcpkgs/CalculiX-cgx/template:17]: I didn't notice any errors with it removed so I didn't think it was necessary to specify

File: srcpkgs/tetgen/template [hunk 26]
```diff
@@ -0,0 +1,26 @@
...
+version=1.5.1
+revision=1
+build_style="gnu-makefile"
+make_cmd="make -C ${pkgname}${version} -f makefile"
+hostmakedepends="tar"
+short_desc="Quality Tetrahedral Mesh Generator and 3D Delaunay Triangulator"
+maintainer="Phicem <phicem@gmx.com>"
+license="AGPL-3.0-or-later"
+homepage="https://www.wias-berlin.de/software/index.jsp?id=TetGen"
+distfiles="http://www.dhondt.de/cgx_2.20.all.tar.bz2"
```
[the remark this answers is not in the document]
Comment (Phicem) [reply srcpkgs/tetgen/template:11]: I will merge tetgen binary into cgx package (see conversation above), so version synchronization will not be needed anymore

File: srcpkgs/CalculiX-cgx/template (hunk 16 above)
Comment (Phicem) [reply srcpkgs/CalculiX-cgx/template:8]: I tried to keep the build-style to "gnu-makefile" instead of overriding do_build(), and it seems to work. Is there a particular reason why you used a custom do_build() function?

File: srcpkgs/tetgen/template (hunk 2 above)
Comment (Phicem) [reply srcpkgs/tetgen/template]: Okay, thank you for the detail. Tetgen is now built directly in CalculiX-cgx package.

**Comment by classabbyamp** [comment]
> I also added a `-O2` flag in FFLAGS of CalculiX-ccx.

it's already in there:

**Comment by Phicem** [comment]
Actually I used a variable named `_FFLAGS` instead of `FFLAGS`, because an xlint warning told me to prefix custom variables with`_`. I did not realize that this change of name would make me lose the benefit of getting default FFLAGS.

I will fix this and use `FFLAGS` directly, but I'm expecting that xlint test will fail. By the way, shouln't we add `FFLAGS` to the list of official variables recognized by xlint?
